### PR TITLE
Bug 1963730: kube-apiserver failed to load SNI cert and key

### PR DESCRIPTION
**Pull request by p0lyn0mial**
The `target directory` holds all files for the current static pod manifest.

As of today, `kube-apiserver-certs` is accessed by at least two processes, namely the `installer pod` and the `cert-syncer`.
Up until now, there was no coordination between these processes that might have lead to many unexpected errors, like https://bugzilla.redhat.com/show_bug.cgi?id=1963730

Moving the cert directory to the `target directory` (revisioned) ensures that these processes won't step each other's toes.

After this PR `/etc/kubernetes/static-pod-resources/kube-apiserver-pod-REVISION/dynamic`
Previously it was `/etc/kubernetes/static-pod-resources/kube-apiserver-certs`

### CODE REVIEW

File: bindata/v4.1.0/kube-apiserver/pod.yaml [hunk 1]
```diff
@@ -238,7 +238,7 @@ spec:
       path: /etc/kubernetes/static-pod-resources/kube-apiserver-pod-REVISION
     name: resource-dir
   - hostPath:
-      path: /etc/kubernetes/static-pod-resources/kube-apiserver-certs
+      path: /etc/kubernetes/static-pod-resources/kube-apiserver-pod-REVISION/kube-apiserver-certs
```
Comment (sttts) [inline bindata/v4.1.0/kube-apiserver/pod.yaml:241]: cert-sync-certs maybe?

Comment (sttts) [reply bindata/v4.1.0/kube-apiserver/pod.yaml:241]: or unrevisioned ?

Comment (sttts) [reply bindata/v4.1.0/kube-apiserver/pod.yaml:241]: or cert-sync

Comment (sttts) [reply bindata/v4.1.0/kube-apiserver/pod.yaml:241]: or dynamic

Comment (sttts) [reply bindata/v4.1.0/kube-apiserver/pod.yaml:241]: @mfojtik join the bike shedding

Comment (p0lyn0mial) [reply bindata/v4.1.0/kube-apiserver/pod.yaml:241]: +1

as of today we have
```
ls /etc/kubernetes/static-pod-resources/kube-apiserver-pod-18/kube-apiserver-certs
configmaps  secrets
```

Comment (p0lyn0mial) [reply bindata/v4.1.0/kube-apiserver/pod.yaml:241]: I like `dynamic`

Comment (p0lyn0mial) [reply bindata/v4.1.0/kube-apiserver/pod.yaml:241]: The name `dynamic` is controlled by the operator. So we would have to change it in the other places as well.

**Comment by p0lyn0mial** [comment]
/hold

for the fake bump

**Comment by p0lyn0mial** [comment]
/hold

as it is not clear if the DR scripts will work

**Comment by p0lyn0mial** [comment]
/close 

closing in favour of https://github.com/openshift/cluster-kube-apiserver-operator/pull/1145

**Pull request closed without merging**